File: LeitorTexto.java
package leitores;

import java.io.*;
import java.awt.image.BufferedImage;
import javax.imageio.ImageIO;

public class LeitorTexto  {
	private FileReader arquivo;
	private BufferedReader texto;
	private File arquivoImagem;
	private String linha;
	private BufferedImage imagem;
	
	public String converteTexto() {
		try {
			int comprimento, largura, posicaoX, posicaoY, R, B, G, pixel;
			
			String caminhoDiretorio = Arquivos.retornaCaminho('t');
			String salvarArquivo = Arquivos.salvarCaminho();
			
			arquivo = new FileReader(caminhoDiretorio);
			texto = new BufferedReader(arquivo);
			arquivoImagem =  new File(salvarArquivo + "/saidaImagem.jpg");
			
			linha = texto.readLine();
			
			String[] textoSplit = linha.split("%");
			comprimento = Integer.parseInt(textoSplit[0]);
			largura = Integer.parseInt(textoSplit[1]);
			
			imagem = new BufferedImage(comprimento, largura, BufferedImage.TYPE_INT_RGB);
			
			while ((linha = texto.readLine()) != null) {
				textoSplit = linha.split(" ");
				
				if (textoSplit[0] == "n") {
					posicaoX = Integer.parseInt(textoSplit[1]);
					posicaoY = Integer.parseInt(textoSplit[2]);

					R = Integer.parseInt(textoSplit[3]);
					G = Integer.parseInt(textoSplit[4]);
					B = Integer.parseInt(textoSplit[5]);
				
					pixel = (R << 16) | (G << 8) | B;
				
					imagem.setRGB(posicaoY, posicaoX, pixel);
				} else if (textoSplit[0] == "c") {
					int posicaoYf;

					posicaoX = Integer.parseInt(textoSplit[1]);
					posicaoY = Integer.parseInt(textoSplit[2]);
					posicaoYf = Integer.parseInt(textoSplit[3]);	

					R = Integer.parseInt(textoSplit[4]);
					G = Integer.parseInt(textoSplit[5]);
					B = Integer.parseInt(textoSplit[6]);	

					pixel = (R << 16) | (G << 8) | B;

					for (int contador = posicaoY; contador <= posicaoYf; contador++) {
						imagem.setRGB(posicaoY, contador, pixel);						
					}		
				}
			}
			
			ImageIO.write(imagem, "jpg", arquivoImagem);
			
			texto.close();
			
			return "A conversão foi realizada com sucesso!";
		} catch(Exception e) {
			return "Não foi possível realizar a operação...";
		}
	}
}

File: LeitorTexto.kt
package leitores

import java.awt.image.BufferedImage
import java.io.BufferedReader
import java.io.File
import java.io.FileReader
import javax.imageio.ImageIO

class LeitorTexto {
    private val arq: Arquivos = Arquivos()

    private var arquivo: FileReader? = null
    private var texto: BufferedReader? = null
    private var arquivoImagem: File? = null
    private var linha: String? = null
    private var imagem: BufferedImage? = null

    fun converteTexto(): String {
        var x: Int
        var y: Int
        var r: Int
        var g: Int
        var b: Int
        var pixel: Int

        try {
            val caminhoDiretorio: String = arq.retornaCaminho("t")
            val salvarArquivo: String = arq.salvarCaminho()

            arquivo = FileReader(caminhoDiretorio)
            texto = BufferedReader(arquivo)
            arquivoImagem = File("$salvarArquivo/saidaImagem.jpg")

            linha = texto!!.readLine()

            var textoSplit = linha!!.split("%").toTypedArray()

            val comprimento = textoSplit[0].toInt()
            val largura = textoSplit[1].toInt();

            imagem = BufferedImage(comprimento, largura, BufferedImage.TYPE_INT_RGB)

            while (texto!!.readLine().also { linha = it } != null) {
                textoSplit = linha!!.split(" ").toTypedArray()

                if (textoSplit[0] == "n") {
                    x = textoSplit[1].toInt()
                    y = textoSplit[2].toInt()

                    r = textoSplit[3].toInt()
                    g = textoSplit[4].toInt()
                    b = textoSplit[5].toInt()

                    pixel = r shl 16 or (g shl 8) or b

                    imagem!!.setRGB(y, x, pixel)
                } else {
                    //TODO Adicionar para múltiplas linhas
                }
            }

            ImageIO.write(imagem, "jpg", arquivoImagem)
            texto!!.close()

            return "A conversão foi realizada com sucesso..."
        } catch (e: Exception) {
            return "Não foi possível realizar a operação. Erro: ${e.stackTraceToString()}"
        }
    }
}